docs/source/gateway.rst

```diff
@@ -0,0 +1,745 @@
+
+.. code-block:: sh
+
+    pip install -e '.[gateway]'
+
+
+Step 2: Set the OpenAI API Key(s) for each provider
+---------------------------------------------------
+The Gateway service needs to communicate with the OpenAI API. To do this, it requires an API key.
+You can create an API key from the OpenAI dashboard.
+
+For this example, we're only connecting with OpenAI. If there are additional providers within the
+configuration, these keys will need to be set as well.
+
+Once you have the key, you can set it as an environment variable in your terminal:
+
+.. code-block:: sh
+
+    export OPENAI_API_KEY=your_api_key_here
+
+This sets a temporary session-based environment variable. For production use cases, it is advisable
+to store this key in the ``.bashrc`` or ``.zshrc`` files so that the key doesn't have to be re-entered upon
+system restart.
+
+Step 3: Create a Gateway Configuration File
+-------------------------------------------
+Next, you need to create a Gateway configuration file. This is a YAML file where you specify the
+routes that the Gateway service should expose. Let's create a file with three routes using OpenAI as a provider: completions, chat, and embeddings.
+
+.. code-block:: yaml
+
+    routes:
+      - name: completions
+        type: llm/v1/completions
+        model:
+          provider: openai
+          name: gpt-3.5-turbo
+          config:
+            openai_api_base: https://api.openai.com/v1
+            openai_api_key: $OPENAI_API_KEY
+
+      - name: chat
+        type: llm/v1/chat
+        model:
+          provider: openai
+          name: gpt-3.5-turbo
+          config:
+            openai_api_base: https://api.openai.com/v1
+            openai_api_key: $OPENAI_API_KEY
+
+      - name: embeddings
+        type: llm/v1/embeddings
+        model:
+          provider: openai
+          name: text-embedding-ada-002
+          config:
+            openai_api_base: https://api.openai.com/v1
+            openai_api_key: $OPENAI_API_KEY
+
+Save this file to a location on the system that is going to be running the `MLflow Gateway` server.
+
+Step 4: Start the Gateway Service
+---------------------------------
+You're now ready to start the Gateway service!
+
+Use the ``mlflow gateway start`` command and specify the path to your configuration file:
+
+.. code-block:: sh
+
+    mlflow gateway start --config-path config.yaml --port {port} --host {host} --workers {worker count}
+
+If you do not specify the host, a localhost address will be used.
+
+If you do not specify the port, port 5000 will be used.
+
+The worker count for gunicorn is defaulted as 2 workers.
+
+Step 5: Access the Interactive API Documentation
+------------------------------------------------
+The `MLflow Gateway` service provides an interactive API documentation endpoint that you can use to explore
+and test the exposed routes. Navigate to http://{host}:{port}/ (or http://{host}:{port}/docs) in your browser to access it.
```

Comment:
This renders weird (the http:// renders as an empty link). Not sure how to fix this.